File: python/task4.py
import math

def index(character):
    Array = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
         'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J',
         'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T',
         'U', 'V', 'W', 'X', 'Y', 'Z']
    index = -1
    for i in range(0,len(Array)):
        if character == Array[i]:
            index = i
            break
    return index

def Encrypt(base,Number):
    output = 0
    for i in range(0,len(Number)):
        output += index(Number[i]) *pow(base,i)
    return output

baseinput = int(input())
userinput = str(input())
Array = []

while userinput != "-1":
    Array.append(Encrypt(baseinput,userinput[::-1]))
    userinput = str(input())

for i in Array:
    print(i)

File: c++/task4.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>
#include <cmath>

using namespace std;
using Arrays = vector<int>;

int BaseTen(int Base, string Number);

int main() {
    int UserInput;
    int BaseInput;
    Arrays Output;

    cin >> BaseInput;
    cin >> UserInput;
    
    while(UserInput != -1)
      Output.push_back(BaseTen(BaseInput, to_string(UserInput)));
      cin >> UserInput;
    }
    
    for(int i : Output){
        cout<<i<<"\n";
    }
    
    return 0;
}

int BaseTen(int Base, string Number) {
    int Num = 0;

    for (int index = Number.length() - 1; index > -1; index--) {
        int digit = Number[index] - '0';
        Num +=  digit * pow(Base, Number.length() - index - 1);
    }
    return Num;
}

